zerver/views/messages.py

```diff
@@ -171,8 +171,13 @@ def by_is(self, query, operand, maybe_negate):
         elif operand == 'unread':
             cond = column("flags").op("&")(UserMessage.flags.read.mask) == 0
             return query.where(maybe_negate(cond))
-        elif operand == 'mentioned' or operand == 'alerted':
-            cond = column("flags").op("&")(UserMessage.flags.mentioned.mask) != 0
+        elif operand == 'mentioned':
+            cond1 = column("flags").op("&")(UserMessage.flags.mentioned.mask) != 0
+            cond2 = column("flags").op("&")(UserMessage.flags.wildcard_mentioned.mask) != 0
+            cond = or_(cond1, cond2)
+            return query.where(maybe_negate(cond))
```

Comment:
May be worth looking into whether this means we need another database index in the style of the one we just added for `mentioned` queries.  I can check what database query gets generated and thus what we need here; same is probably true for `is:alerted`.

Also, I guess `is:alerted` was just broken before?